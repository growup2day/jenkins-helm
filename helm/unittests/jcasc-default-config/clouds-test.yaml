templates:
  - config-jcasc.yaml
tests:
  - it: should include openshift cloud in default config script when configured
    set:
      domain: dummy.domain
      JCasC:
        defaultConfig: true
        openshiftClouds:
          cs1-dev:
            serverUrl: "https://api.cs1-dev.nz.service.test:6443"
            namespace: "{{ .Release.Namespace }}"
            credentialsId: "cs1-dev-cred"
            webSocket: true
            defaultsProviderTemplate: "default-jenkins-slave"
            simpleTemplates:
              commonPodAttributes:
                idleMinutes: 30
                instanceCap: 8
                yamlMergeStrategy: "override"
              commonContainerAttributes:
                resourceRequestCpu: "200m"
                resourceRequestMemory: "500Mi"
                resourceLimitCpu: "2"
                resourceLimitMemory: "4Gi"
              commonEnvVars:
                TZ: "Pacific/Auckland"
                http_proxy: ""
                https_proxy: ""
              templates:
                default-jenkins-slave:
                  label: "openshift"
                  image:
                    path: "openshift/jenkins-slave-base-ocp-anz:v4.7.2"
                maven-jenkins-slave: # minimal definition
                  label: "openshift maven"
                  image:
                    path: "openshift/jenkins-slave-maven-ocp-anz:v4.7.2"
                nodejs-jenkins-slave: # testing for common attributes override
                  label: "openshift nodejs"
                  image:
                    path: "openshift/jenkins-slave-nodejs-ocp-anz:v4.7.2"
                  useGoInit: true
                  podAttributes:
                    idleMinutes: 10
                  containerAttributes:
                    resourceLimitCpu: "4"
                    workingDir: "/tmp"
                  envVars:
                    https_proxy: "test.proxy"
                    CONCURRENT_BUILD_TASKS: "2"
          cs2-dev:
            serverUrl: "https://api.cs2-dev.nz.service.test:6443"
            namespace: "{{ .Release.Namespace }}"
            webSocket: false
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-clouds.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-clouds.yaml]
  - it: should reference nodePort when the Jenkins master is configured with it
    set:
      domain: dummy.domain
      jnlpService:
        enabled: true
        nodePort: 33123
      JCasC:
        defaultConfig: true
        openshiftClouds:
          cs1-dev:
            serverUrl: "https://api.cs1-dev.nz.service.test:6443"
            namespace: "{{ .Release.Namespace }}"
            credentialsId: "cs1-dev-cred"
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-clouds.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-clouds.yaml]
  - it: should include blank cloud in default config script when not configured
    set:
      domain: dummy.domain
      JCasC:
        defaultConfig: true
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-clouds.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-clouds.yaml]
  - it: should not include cloud config in default config scripts when there is custom config script for the section
    set:
      JCasC:
        defaultConfig: true
        configScripts:
          custom-override: |
            jenkins:
              clouds:
                - kubernetes:
                    name: cs1-dev
                    serverUrl: "https://api.cs1-dev.nz.service.test:6443"
    asserts:
      - isNull:
          path: data.[jcasc-default-config-clouds.yaml]
      - isNotNull:
          path: data.[custom-override.yaml]
