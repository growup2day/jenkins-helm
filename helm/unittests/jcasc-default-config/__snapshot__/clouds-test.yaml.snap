should include blank cloud in default config script when not configured:
  1: |
    |-
      jenkins:
        # Manage Nodes and Clouds > Configure Clouds
        clouds: []
should include openshift cloud in default config script when configured:
  1: |
    |-
      jenkins:
        # Manage Nodes and Clouds > Configure Clouds
        clouds:
          - kubernetes:
              name: cs1-dev
              serverUrl: "https://api.cs1-dev.nz.service.test:6443"
              namespace: "NAMESPACE"
              credentialsId: "cs1-dev-cred"
              webSocket: true
              jenkinsUrl: "https://RELEASE-NAME-jenkins-NAMESPACE.dummy.domain"
              defaultsProviderTemplate: "default-jenkins-slave"
              templates:
                - name: "default-jenkins-slave"
                  label: "openshift"
                  nodeSelector: "beta.kubernetes.io/os=linux"
                  serviceAccount: "RELEASE-NAME-jenkins"
                  idleMinutes: 30
                  instanceCap: 8
                  yamlMergeStrategy: override
                  containers:
                    - name: "jnlp"
                      command: "/usr/local/bin/run-jnlp-client"
                      args: "-webSocket ^${computer.jnlpmac} ^${computer.name}"
                      image: "image-registry.openshift-image-registry.svc:5000/openshift/jenkins-slave-base-ocp-anz:v4.7.2"
                      resourceLimitCpu: "2"
                      resourceLimitMemory: 4Gi
                      resourceRequestCpu: 200m
                      resourceRequestMemory: 500Mi
                      envVars:
                        - envVar:
                            key: "TZ"
                            value: "Pacific/Auckland"
                        - envVar:
                            key: "http_proxy"
                            value: ""
                        - envVar:
                            key: "https_proxy"
                            value: ""
                - name: "maven-jenkins-slave"
                  label: "openshift maven"
                  nodeSelector: "beta.kubernetes.io/os=linux"
                  serviceAccount: "RELEASE-NAME-jenkins"
                  idleMinutes: 30
                  instanceCap: 8
                  yamlMergeStrategy: override
                  containers:
                    - name: "jnlp"
                      command: "/usr/local/bin/run-jnlp-client"
                      args: "-webSocket ^${computer.jnlpmac} ^${computer.name}"
                      image: "image-registry.openshift-image-registry.svc:5000/openshift/jenkins-slave-maven-ocp-anz:v4.7.2"
                      resourceLimitCpu: "2"
                      resourceLimitMemory: 4Gi
                      resourceRequestCpu: 200m
                      resourceRequestMemory: 500Mi
                      envVars:
                        - envVar:
                            key: "TZ"
                            value: "Pacific/Auckland"
                        - envVar:
                            key: "http_proxy"
                            value: ""
                        - envVar:
                            key: "https_proxy"
                            value: ""
                - name: "nodejs-jenkins-slave"
                  label: "openshift nodejs"
                  nodeSelector: "beta.kubernetes.io/os=linux"
                  serviceAccount: "RELEASE-NAME-jenkins"
                  idleMinutes: 10
                  instanceCap: 8
                  yamlMergeStrategy: override
                  containers:
                    - name: "jnlp"
                      command: "/usr/bin/go-init"
                      args: "-main \"/usr/local/bin/run-jnlp-client -webSocket ^${computer.jnlpmac} ^${computer.name}\""
                      image: "image-registry.openshift-image-registry.svc:5000/openshift/jenkins-slave-nodejs-ocp-anz:v4.7.2"
                      resourceLimitCpu: "4"
                      resourceLimitMemory: 4Gi
                      resourceRequestCpu: 200m
                      resourceRequestMemory: 500Mi
                      workingDir: /tmp
                      envVars:
                        - envVar:
                            key: "CONCURRENT_BUILD_TASKS"
                            value: "2"
                        - envVar:
                            key: "TZ"
                            value: "Pacific/Auckland"
                        - envVar:
                            key: "http_proxy"
                            value: ""
                        - envVar:
                            key: "https_proxy"
                            value: "test.proxy"
          - kubernetes:
              name: cs2-dev
              serverUrl: "https://api.cs2-dev.nz.service.test:6443"
              namespace: "NAMESPACE"
              jenkinsTunnel: "RELEASE-NAME-jenkins-jnlp.NAMESPACE.svc.cluster.local:50000"
              jenkinsUrl: "https://RELEASE-NAME-jenkins-NAMESPACE.dummy.domain"
should reference nodePort when the Jenkins master is configured with it:
  1: |
    |-
      jenkins:
        # Manage Nodes and Clouds > Configure Clouds
        clouds:
          - kubernetes:
              name: cs1-dev
              serverUrl: "https://api.cs1-dev.nz.service.test:6443"
              namespace: "NAMESPACE"
              credentialsId: "cs1-dev-cred"
              jenkinsTunnel: "RELEASE-NAME-jenkins-NAMESPACE.dummy.domain:33123"
              jenkinsUrl: "https://RELEASE-NAME-jenkins-NAMESPACE.dummy.domain"
